src/hal/src/format.rs

```diff
@@ -569,10 +569,15 @@ impl Format {
     ///
     /// Returns `None` if format is `Undefined`.
     pub fn base_format(self) -> BaseFormat {
-        assert!(NUM_FORMATS > self as usize);
+        assert!(self as usize != 0 && NUM_FORMATS > self as usize);
```

Comment:
Is there still a way to do this now that `Undefined` has been removed?

Reply:
forwarding to @msiglreith

Reply:
Might be. For portability a crash will most likely be fine but probably nice to have an additional check here.